test/ecto/query/planner_test.exs

```diff
@@ -16,9 +16,49 @@ defmodule Ecto.Query.PlannerTest do
       field :temp, :string, virtual: true
       field :posted, :naive_datetime
       field :uuid, :binary_id
+      field :special, :boolean
       belongs_to :post, Ecto.Query.PlannerTest.Post
       has_many :post_comments, through: [:post, :comments]
     end
+
+    def special() do
+      from comment in __MODULE__,
+        where: comment.special
+    end
+  end
+
+  defmodule Author do
```

Comment:
We prefer to using the existing modules instead of adding new modules, otherwise it becomes impossible to manage all of the schemas in this file. Would that be possible?

Reply:
I definitely can, but it will be a bit strange modeling the join association that way. If I could add one schema for the join then that would be enough to do it cleanly.

Reply:
Sure! Just add the minimum necessary. :)